Merge Foxy devel to Main

WinstonHChou opened the pull request.

WinstonHChou committed:
Update racer.repos

change livox repo to original ones.
add pointcloud_to_laserscan pkg

WinstonHChou committed:
add livox_pointcloud_to_laserscan_launch.py

WinstonHChou committed:
Update Makefile

add source-livox

WinstonHChou committed:
Update Makefile

remove source-livox
"source" does not work in Makefile

WinstonHChou committed:
Update racer.repos

WinstonHChou committed:
Update racer.repos

WinstonHChou committed:
add rviz to basestation_launch

WinstonHChou committed:
Add files via upload

WinstonHChou committed:
Update package.xml

WinstonHChou committed:
Update CMakeLists.txt

WinstonHChou committed:
add yaml files to param

WinstonHChou committed:
add launch files to launch

WinstonHChou committed:
Update joy.launch.py

WinstonHChou committed:
Update joy.launch.py

WinstonHChou committed:
Update joy.param.yaml

WinstonHChou committed:
Update racer.repos

WinstonHChou committed:
Create lidar_utils.repos

WinstonHChou committed:
Update all_components.launch.py

WinstonHChou committed:
add support to all_nodes.launch.py, add support to launch livox_MID360 in basestation_launch, change joy.param.yaml settings

WinstonHChou committed:
add support to pointcloud_to_laserscan

WinstonHChou committed:
minor fix

WinstonHChou committed:
minor fix

Commit:
add bash script source_ros2.sh

WinstonHChou committed:
Update all_components.launch.py

WinstonHChou committed:
Update all_nodes.launch.py

Commit:
add support to rmw-cyclonedds

Commit:
minor fix for livox_pointcloud_to_laserscan.launch.py

Commit:
minor fix

Commit:
add support to rosbride-suite

Commit:
build_ros2_pkg() auto-complete basestation_launch, vesc_odom

Commit:
start working on Dockerfile.f1tenth_ros

WinstonHChou committed:
Create publish.yml

WinstonHChou committed:
workflows added

WinstonHChou committed:
workflows fixing

WinstonHChou committed:
workflows update to node20

WinstonHChou committed:
update workflows

WinstonHChou committed:
workflow fixing

WinstonHChou committed:
workflow fixing

WinstonHChou committed:
workflow fixing

WinstonHChou committed:
workflow fixing

WinstonHChou committed:
workflow fixing

WinstonHChou committed:
workflow fixing

WinstonHChou committed:
docker build cloud testing

WinstonHChou committed:
failed testing

WinstonHChou committed:
minor fix

WinstonHChou committed:
workflows: activate push -> release

Commit:
fix auto-complete flag in source_ros2.sh

Commit:
add support to /odom

Commit:
add slam-toolbox support

WinstonHChou committed:
Update README.md

WinstonHChou committed:
Update docker.sh

WinstonHChou committed:
urdf fixed

WinstonHChou committed:
Update Dockerfile.f1tenth_ros

WinstonHChou committed:
Update Dockerfile.f1tenth_ros

WinstonHChou committed:
lidar inverted as default

WinstonHChou committed:
create planner pkg, move slam-toolbox launch/yaml files to `planner` pkg, support variable numbers of arguments on `build_ros2_pkg`

WinstonHChou committed:
create `sensors` pkg, move livox-lidar related to `sensors` pkg

WinstonHChou committed:
sorting out files, most of the settings can be adjusted in yaml files

Commit:
create  pkg, move slam-toolbox to  pkg

Commit:
move robot_localization to state_estimation pkg

Commit:
map_server.launch.py added to state_estimation pkg

Commit:
add amcl_server localization support

WinstonHChou committed:
minor fix

WinstonHChou committed:
minor fix, regarding map_server

Commit:
WIP: planner pkg (please do not build)

Commit:
WIP: `controller/`

WinstonHChou committed:
fixing map_server, amcl_server

WinstonHChou committed:
add support to `nav2_map_server`, `nav2_amcl`

Commit:
WIP: `planner` pkg, initial working, sort out avalible nav2 plugins for ros2 foxy

WinstonHChou committed:
WIP: `planner` pkg, change to `smac_planner` plugin, add `Groot` support to `Dockerfile.f1tenth_ros`

WinstonHChou committed:
WIP: `planner` pkg, add `stvl_layer` support to `local costmap`

WinstonHChou committed:
WIP: `planner` pkg, `global costmap` fine-tuning

WinstonHChou committed:
minor fix

WinstonHChou committed:
WIP: `planner` pkg, npvl added to `global costmap`, path planning works

Commit:
add `twist_to_ackermann` pkg, able to convert `/cmd_vel` to `/drive`

WinstonHChou committed:
minor fix, `speed` from `/cmd_vel` is too slow, finding a way to adjust that

WinstonHChou committed:
update ackermann_mux priorities

WinstonHChou committed:
WIP: `planner` pkg, yaml_filename changed to: `controller_server.yaml`

WinstonHChou committed:
fixing bt_nav and twist_to_ack

WinstonHChou committed:
fixes `AngleToServo` for vesc driver

WinstonHChou committed:
add new param of `default_steering_direction` in `tta` pkg

WinstonHChou committed:
add support to dockerfile for sim in x86, might need a smaller base image

WinstonHChou committed:
WIP: `foxy_addon_behavior_tree` pkg (do not build)

WinstonHChou committed:
WIP: `planner` pkg, added static, dynamic follow behavior

WinstonHChou committed:
moved `planner` pkg to `path_planning/planner/`

WinstonHChou committed:
minor update, prepare for `gb_optimizer`

Commit:
WIP: `gb_optimizer_nav2` pkg, add `f110_msgs` pkg for ros 2

Commit:
minor fix on structure

Commit:
a large update regarding `gb_optimizer_nav2` building venv

Commit:
remove gb_opt_env from git, instead, uses a temporary venv when building `gb_optimizer_nav2`

Commit:
minor undo, instead of moving gb_optimizer from ros, move the ros2 branch

Commit:
WIP: `global planner` pkg, it's a little messy

Commit:
fixing global trajectory publisher

Commit:
remove `gb_optimizer_nav2`

WinstonHChou committed:
fixes dependency issues with `global_planner` pkg

WinstonHChou committed:
WIP: `global_planner` pkg, add support to both `*.pgm` and `*.png` map file extension

WinstonHChou committed:
fixes dependencies to `global_planner` entirely

WinstonHChou committed:
WIP: `global_planner` pkg, added `global_path_publisher.launch.py`

WinstonHChou committed:
minor fix

WinstonHChou committed:
add a parent folder `utilities/` to `f110_msgs` pkg

WinstonHChou committed:
update rviz configuration

WinstonHChou committed:
minor fix, starts working on ekf

WinstonHChou committed:
minor fix, restructure tf tree: `odom` -> `base_footprint` -> `base_link` -> ...

WinstonHChou committed:
WIP: `state_estimation` pkg, capable of using ekf node

WinstonHChou committed:
update dockerfiles

WinstonHChou committed:
minor fix

WinstonHChou committed:
minor fix

WinstonHChou committed:
minor fix on GitHub workflow

WinstonHChou committed:
minor fix

WinstonHChou committed:
minor fix

WinstonHChou committed:
fix typo

WinstonHChou committed:
`Dockerfile.f1tenth_x86` Test Build

WinstonHChou committed:
dockerfile `global_planner` dependencies fix

WinstonHChou committed:
update `publish.yml`, officially rename branch `livox_devel` to `foxy_devel`

WinstonHChou committed:
minor test after changing branch name

WinstonHChou committed:
add $ROS_DISTRO to package name

WinstonHChou committed:
add --no-install-recommends

WinstonHChou committed:
put $ROS_DISTRO to image tags, fix main repo in docker images

WinstonHChou committed:
minor fix on dockerfiles

WinstonHChou committed:
fix a bit for custom terminal setting in bashrc

WinstonHChou committed:
support `r2fo_laser_odometry`

WinstonHChou committed:
minor update

WinstonHChou committed:
move `sensors` pkg to `src/sensors/`, move `state_estimation` pkg to `src/state_estimation`

WinstonHChou committed:
WIP: `ekf.launch.py`

WinstonHChou committed:
fixes dockerfiles

WinstonHChou committed:
minor fix

chenyenru committed:
Add notes from Winston

chenyenru committed:
updated docker_sim.sh

chenyenru committed:
added commens of nav2 on node_config.yaml

chenyenru committed:
Merge branch 'foxy_devel' of github.com:Triton-AI/robocar into foxy_devel

WinstonHChou committed:
WIP: `state_estimation` pkg,  able to manually select the sensor sources we want to do sensor fusion by just turning on or off in `ekf_select.yaml`.

WinstonHChou committed:
Remapping topic names: `/scanner/scan` -> `/ego_racecar/scan`, `/robot_description` -> `/ego_racecar/robot_description`, `/joint_states` -> `/ego_racecar/joint_states`

WinstonHChou committed:
add map `lab_0525`

WinstonHChou committed:
Remapping topic names: `/drive` -> `/ego_racecar/drive`, `/teleop` -> `/ego_racecar/teleop`

WinstonHChou committed:
add `apt-get install xauth` to dockerfiles

WinstonHChou committed:
Merge branch 'foxy_devel' of https://github.com/Triton-AI/robocar into foxy_devel

WinstonHChou committed:
WIP: work on adding `nav2_waypoint_follower` support

WinstonHChou committed:
switch extension for urdf file, `*.urdf` to `*.xacro`

WinstonHChou committed:
rename `ucsd_racercar` to `ucsd_racecar`

WinstonHChou committed:
`node_pkg_locations`: `odom_ekf` renamed to `ekf`

WinstonHChou committed:
minor fix r2fo

raymondsong00 commented on src/launch/basestation_launch/ros_data/maps/lab_test/lab_0525/lab_0525.pgm:
Add .pgm to your gitignore

WinstonHChou replied:
Can we add gitignore only at a directory that contains self-testing map? So that at least, people can put their map data there if they don't want to push them back to the repo.

raymondsong00 replied:
Yea you can just add the directory to the gitignore. What you can do to keep that directory is to add a blank .gitkeep so that git can keep the directory there. You don't need to name it .gitkeep but that is just a convention people do.

WinstonHChou replied:
Added `src/launch/basestation_launch/ros_data/maps/self_test/` to save local repo's map data.
Added explanation and naming rules to `ros_data/maps/README.md`.
Added `src/launch/basestation_launch/ros_data/maps/self_test/*/*` to `.gitignore`.

raymondsong00 commented on .vscode/c_cpp_properties.json:
You can add .vscode to the gitignore.

raymondsong00 commented on src/launch/basestation_launch/ros_data/frames/frames.gv:
Add path to gitignore.

WinstonHChou committed:
update gitignore

WinstonHChou committed:
create new `maps/self_test/` directory to save local repo map data

WinstonHChou commented:
If no one has issue with the merge, it'll be merged tonight.
@Triton-AI/f1tenth-robocar

chenyenru commented on .gitignore:
Can we include these too?
```
# Executables
*.exe
*.out
*.app

# macos
.DS_Store

# py
*.pyc

#vscode
.vscode
.vscode/build
.vscode/log
.vscode/c_cpp_properties.json
```

WinstonHChou replied:
of course. But can you briefly explain what `.vscode/c_cpp_properties.json` contains?

chenyenru replied:
I think that stores your VSCode's C++ config.

WinstonHChou replied:
do you think keeping it helps using vscode developing?
personally, i do add some includepath when developing in c++

chenyenru replied:
you're right, the c++ config will help a lot

WinstonHChou replied:
i will add others to gitignore, but leave vscode the same for now.

chenyenru replied:
sounds good

chenyenru commented on docker.sh:
```diff
@@ -0,0 +1,19 @@
+    --volume='/home/jetson/.Xauthority:/root/.Xauthority:rw' \
+    --volume='/tmp/.X11-unix/:/tmp/.X11-unix' \
+    --volume='/home/jetson/projects/robocar:/home/jetson/projects/robocar' \
+    ghcr.io/ucsd-ecemae-148/donkeycontainer:ros
```
should we use `ghcr.io/triton-ai/robocar:foxy-x86` here?

WinstonHChou replied:
no that's for arm arch

WinstonHChou replied:
should find some time to work on the arm arch image

WinstonHChou committed:
update `.gitignore`

raymondsong00 commented on source_ros2.sh:
```diff
@@ -0,0 +1,79 @@
+	colcon build --packages-ignore livox_ros_driver2 livox_sdk2 f1tenth_nav2_ctrl_plugin foxy_addon_behavior_tree --cmake-args -DCMAKE_BUILD_TYPE=Debug
+	source install/setup.bash
+}
+
+function build_ros2_pkg()
+{
+	cd ${WORKSPACE}
+	for arg in "$@"
+	do	
+		if [[ $arg == "livox_ros_driver2" ]] || [[ $arg == "livox_sdk2" ]]
+		then
+		echo "Don't try to colcon build livox packages"
+		return
+		fi
+		if [[ $arg == "gb_opt" ]]
+		then
+		make gb_opt
+		return
+		fi
+		if [[ $arg == "sick_scan_xd" ]]
+		then
+		make sick-driver
+		return
+		fi
+	done
+	colcon build --packages-select $@ --cmake-args -DCMAKE_BUILD_TYPE=Debug
+	source install/setup.bash
+}
+complete -W "basestation_launch vesc_odom sensors planner global_planner state_estimation" build_ros2_pkg
+
+function rmw_switch()
+{
+	if [[ $1 == "cyclonedds" ]]
+	then
+		export RMW_IMPLEMENTATION=rmw_cyclonedds_cpp
+	else
+		export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+	fi
+}
+complete -W "cyclonedds fastrtps" rmw_switch
```
A lot of this stuff can be put into the Makefile instead of having separate scripts.

WinstonHChou replied:
the reason i used bash script in the first place is that makefile command creates a subterminal to execute commands, which causes not able to source ros environment for the existing terminal.

raymondsong00 commented on scripts/bashrc.sh:
Not sure if you need this but you can keep if you want.

WinstonHChou replied:
I like the customized terminal

raymondsong00 commented on src/actuator/twist_to_ackermann/CMakeLists.txt:
```diff
@@ -0,0 +1,75 @@
+### find dependencies
+### ------------------
+
+find_package(ament_cmake REQUIRED)
```
if you use `ament_cmake_auto` you don't need `ament_cmake`.

raymondsong00 commented on src/actuator/twist_to_ackermann/src/twist_to_ackermann_node.cpp:
```diff
@@ -0,0 +1,116 @@
+    {
+        ackermann_msgs::msg::AckermannDriveStamped out{};
+        out.header.frame_id = this->frame_id;
+        out.header.stamp = this->current_time;
```
Use `this->get_clock()->now()` you don't need the current_time or the timer if you are just populating the timestamp.

raymondsong00 commented on src/controller/f1tenth_nav2_ctrl_plugin/include/f1tenth_nav2_ctrl_plugin/regulated_pure_pursuit_controller.hpp:
Can this pure pursuit controller be added through vcs imports? This is a lot of copied code that is from another repo.

WinstonHChou replied:
yes. it's still in development. I copied them from nav2 newest update because nav2 doesn't support ros2 foxy anymore. My thought was to add new features as addition plugins and build them from source.

raymondsong00 replied:
You should make a separate repo for the nav2 stuff and have it under external. The point of external is to house all external code from elsewhere. You can still push and pull to that repo as long as you go to that subdirectory.

WinstonHChou replied:
plan to create a repo for triton ai nav2 plugins

raymondsong00 commented on src/launch/basestation_launch/launch/joy.launch.py:
```diff
@@ -1,19 +1,30 @@
 def generate_launch_description():
-    joy_teleop_config = os.path.join(
-        get_package_share_directory('basestation_launch'),
+    print(get_package_share_directory('joy_teleop'))
```
Remove print statement.

raymondsong00 commented on src/launch/basestation_launch/launch/rosbag_launch.launch.py:
Is this launch file necessary? You are also importing a bunch of things that you aren't using at all.

WinstonHChou replied:
i import it because I know ros bag will be useful in future updates. But it's not be developed currently.

raymondsong00 commented on src/launch/basestation_launch/param/car_config.yaml:
I see you brought back Dominic's launching convention/files. You can keep it for now but I recommend changing this to environment files later on that select which launch files are used instead of this.

WinstonHChou replied:
what does this means? Is there any example that i can check?

raymondsong00 replied:
You can create a file like racer.env and use something like https://github.com/sloria/environs and read the racer.env file in your launch files. Then you have something like if statements about the car type and which sensors to launch. Also makes it so you don't have to build each time you want to change your sensors or launch files.

WinstonHChou replied:
dicuss whether to use *.env files to manage different robocar profiles

raymondsong00 commented on src/path_planning/global_planner/README.md:
Is this your implementation of TUM's global race trajectory optimization into ROS?

WinstonHChou replied:
i copy the ForzaETH Zurich's ros2 implementation. And modify it to adapt our stack.
Is there anything I should be aware of?
I will add reference when documenting the dependencies.

WinstonHChou replied:
Please check the commit 92db768a5d161c523a34f775644a2f050403bd74

raymondsong00 replied:
Please put it under src/external/planning/ instead of src/planning. Put your own code in src/ and other repository code in external/ so that you don't track other repos in this repo.

WinstonHChou replied:
> Please put it under src/external/planning/ instead of src/planning. Put your own code in src/ and other repository code in external/ so that you don't track other repos in this repo.

`global_planner` is not a individual repo but inside a full race_stack repo.
i only need a part of it, so i port it and do some customization.

WinstonHChou replied:
moved global planner to [global_race_planner](https://github.com/Triton-AI/global_race_planner)

raymondsong00 commented on src/path_planning/global_planner/global_planner/global_racetrajectory_optimization:
Either use vcs imports or git submodules.

WinstonHChou replied:
there is `repos/racer.repos` contains the vcs information

raymondsong00 commented on src/path_planning/global_planner/package.xml:
In general, I would remove other people's code in your own commit and add it to the vcs imports. If you do need to make changes to their repo, make a fork in the TritonAI org and vcs import the fork. Otherwise, you have the reviewers looking at code that is external to the repo which is separately from this one.

WinstonHChou replied:
i see. i will made the changes.
FYI, the only outside repo that i did not put under `src/external/*` is the TUM global_racetrajectory_optimization

raymondsong00 replied:
Please move it to src/external.

WinstonHChou replied:
agreed, done

raymondsong00 commented on src/state_estimation/state_estimation/package.xml:
```diff
@@ -0,0 +1,29 @@
+  <depend>std_msgs</depend>
+  <depend>geometry_msgs</depend>
+  <depend>ackermann_msgs</depend>
+
```
Add the dependencies needed for nav2 and cartographer etc.

raymondsong00 reviewed (changes requested):
Please use vcs imports for external repositories and fork repos that you need to make changes to. Otherwise, I am reviewing code that has been reviewed by other organizations. If you want to develop by cloning in, that is fine but do not push other people's repos/code as your own (I know you did not do it intentionally) and use either submodules or vcs instead.

WinstonHChou committed:
Change `f1tenth_gym`, `f1tenth_gym_ros` using Triton-AI forked repos, and update dockerfiles for the matter

WinstonHChou committed:
WIP: added `sim.launch.py`, `sim.yaml` to `basestation_launch` pkg (remains unsupported)

WinstonHChou committed:
add `nav2_msgs`, `cartographer_ros_msgs` dependencies to `src/state_estimation/state_estimation/package.xml`

WinstonHChou committed:
update `.gitignore` to meet PR #4

WinstonHChou committed:
Modified `twist_to_ackermann` pkg to meet PR #4

WinstonHChou committed:
update `joy.launch.py` to meet PR #4

WinstonHChou committed:
add ForzaETH reference to `global_planner` pkg to meet PR #4

WinstonHChou committed:
update urdf file, add `ego_racecar/` as prefix every frame, except `odom`, `map`, and frames from sensors themselves

WinstonHChou committed:
other config files that contains `base_link` or `base_footprint`, which are renamed to `ego_racecar/base_link` or `ego_racecar/base_footprint`

WinstonHChou committed:
change the format of scan topic in `pointcloud_to_laserscan_launch.py` and `slice_frame_info.yaml`

WinstonHChou committed:
WIP: simulator devel, add `sim.launch.py` and `sim.yaml` to `basestation_launch` pkg

WinstonHChou committed:
fix typo in `Dockerfile.f1tenth_x86`

WinstonHChou committed:
fix source_ros2 while docker build

WinstonHChou commented:
> Please use vcs imports for external repositories and fork repos that you need to make changes to. Otherwise, I am reviewing code that has been reviewed by other organizations. If you want to develop by cloning in, that is fine but do not push other people's repos/code as your own (I know you did not do it intentionally) and use either submodules or vcs instead.

@raymondsong00 Please check comments

WinstonHChou committed:
Update `Dockerfile.f1tenth_arm`, split x86 and arm arch's building workflows

WinstonHChou committed:
update arm arch workflow

WinstonHChou committed:
update arm arch workflow

WinstonHChou committed:
update workflows triggers

WinstonHChou committed:
update workflows triggers, add fix to arm dockerfile

WinstonHChou committed:
typo of where to put the fix to arm dockerfile

WinstonHChou committed:
Update `dynamic_follow_behavior.xml`

WinstonHChou committed:
add a parent dir `behavior_trees/` to behavior tree files in `planner` pkg

WinstonHChou committed:
add waypoint_detection.py to `planner` pkg, serving as an interface between global and local planner

WinstonHChou committed:
add `global_waypoints_detection.launch.py` and `global_wpnts_detection.yaml` to `planner` pkg, make waypoint detection algorithm simliar to pure-pursuit

WinstonHChou committed:
arm dockerfile update, using `dustynv/opencv:r35.4.1` as base image

WinstonHChou committed:
comment out `FROM cv2 as dl_gpu` in arm dockerfile

WinstonHChou committed:
comment out `FROM dl_gpu as utils`, `FROM dl_gpu as utils` in arm dockerfile

WinstonHChou committed:
fix missing dependencies for arm docker image to install tensorflow

WinstonHChou committed:
change `libhdf5-100` to `libhdf5-103` in arm dockerfile

WinstonHChou committed:
add install `libopenblas-dev` to arm dockerfile

WinstonHChou committed:
add `&& rm -rf /var/lib/apt/lists/* \ && apt-get clean` to dockerfiles

WinstonHChou committed:
add `source /opt/ros/foxy/setup.bash` before `rosdep init` in arm dockerfile

WinstonHChou committed:
comment `source /opt/ros/foxy/setup.bash && \
rosdep init` in arm dockerfile

WinstonHChou committed:
change base image to `dustynv/l4t-ml:r35.4.1` in arm dockerfile

WinstonHChou committed:
change base image to `dustynv/opencv:r35.4.1` in arm dockerfile, `l4t-ml:r35.4.1` didn't work on github actions

WinstonHChou committed:
add free disk space to arm workflow

WinstonHChou committed:
add ARG `ROS_DISTRO` to arm dockerfile

WinstonHChou committed:
add ARG `BRANCH_NAME` to arm dockerfile

WinstonHChou committed:
add ` --fix-missing` after `apt-get update`, when installing ros2 related packages

WinstonHChou committed:
change base image to `dustynv/ros:foxy-desktop-l4t-r35.4.1` in arm dockerfile

WinstonHChou committed:
change back to `dustynv/opencv:r35.4.1`, note that ros in dustynv images is built-from-source

WinstonHChou committed:
add `apt-mark hold libopencv*
` to arm dockerfile, and uncomment ros2 installation

WinstonHChou committed:
add `apt-mark hold manual *opencv*` to try prevent ROS install opencv

WinstonHChou committed:
add dummy opencv pkg

WinstonHChou committed:
comment `apt-get -y dist-upgrade` in dockerfile arm

WinstonHChou committed:
remove `make sim` in dockerfile arm

WinstonHChou committed:
add `source /opt/ros/$ROS_DISTRO/setup.bash && \
    rosdep init` in dockerfile arm

WinstonHChou committed:
moved `sensors` and `state_estimation` pkg to upper level of directories

WinstonHChou committed:
delete `global_planner` , and move `planner` pkg to upper level directory

WinstonHChou committed:
delete controller plugin, future plan: create triton-ai nav2 plugins

WinstonHChou committed:
update bash scripts along with structure changes

WinstonHChou committed:
add global_pathpublisher and global_planner launch files to `planner` pkg

WinstonHChou committed:
add `make racer` as defaut step of setup

WinstonHChou reviewed (commented):
I've moved all codes from other organizations out of this repo

WinstonHChou committed:
add `custom_msg_MID360_launch.py` for developing fast_lio

WinstonHChou committed:
add `FAST_LIO` as one of the repos of `racer.repos`

chenyenru committed:
feat: Added AutoDrive Simulator Dockerfile and entry point script

Helpful when developing racestack with AutoDrive Simulator

chenyenru committed:
feat: Merge f1tenth_x86 and AutoDrive Devkit Dockerfiles

Introduces a new Dockerfile that combines the dependencies and configurations from both the f1tenth_x86 and AutoDrive Devkit Dockerfiles. The resulting Docker container will include all necessary dependencies to run the AutoDrive Devkit. Additionally, the AutoDrive Devkit repository will be cloned into the `src` directory within the container.

Source of AutoDrive Devkit: https://github.com/AutoDRIVE-Ecosystem/AutoDRIVE-F1TENTH-Sim-Racing/releases/tag/2024-cdc

WinstonHChou committed:
add fast_lio launch to `sensors` pkg

WinstonHChou committed:
add fast_lio node to basestation_launch

chenyenru committed:
refactor: Integrate AutoDrive Devkit support into `f1tenth_x86` image

WinstonHChou committed:
add execute `custom_msg_MID360_launch.py` in fast_lio launch

WinstonHChou committed:
remap fast_lio topic: '/Odometry' to '/odom_FAST_LIO'

chenyenru committed:
feat: Add run script for autodrive simulator in docker container

WinstonHChou committed:
add fast_lio to ekf_select

WinstonHChou committed:
add `livox_to_pointcloud2` to `lidar_utils.repos` for converting custom livox msg to ros2 pointcloud2

WinstonHChou committed:
for `planner`, set topic of `NP_voxel_layer` to `/livox/points`

WinstonHChou committed:
For `livox_to_pointcloud2` usage, remap '/livox2/lidar' to '/livox/lidar'

WinstonHChou committed:
modified rviz `Sensors/PointCloud2` topic to `/livox/points`

WinstonHChou committed:
set `urdf_related_param.yaml` to ucsd-blue profile

chenyenru committed:
feat: Add autodrive devkit launch file in basestation_launch to support autodrive simulator in robocar racestack

WinstonHChou committed:
dev: `cartographer`

chenyenru committed:
feat: Added ackermann_odom dependency and  ackermann_odom_node launching in autodrive_sim.launch.py

chenyenru committed:
[Feature] vcs import ackermann odom in the docker container

chenyenru committed:
[Refactoring] Temporarily commenting out nav2 nodes in autodrive_sim.launch.py

chenyenru committed:
[BugFix] Update Dockerfile for vcs import and update autodrive.repos with the correct URL

chenyenru committed:
[BugFix] Resolve missing livox dependency for fast lio

Moved` make livox-driver` before `make racer` so that `livox-driver` is there as a dependency for `fast lio`

chenyenru committed:
[Refactoring] Remove autodrive simulator and dockerfile since we'll be pulling image from docker hub

chenyenru committed:
[Refactor] Remove autodrive_simulator.sh

WinstonHChou committed:
update docker_sim.sh script, includes custom xauth path

chenyenru committed:
[BugFix] Comment out gdb prefix for ackermann_odom

WinstonHChou committed:
[Feat] add `autodrive_sim.launch.py` to node_config.yaml

WinstonHChou committed:
add `autodrive_racecar.xacro`

WinstonHChou committed:
[Feat] Topics Remap: lidar and camera

WinstonHChou committed:
[Rename] autodrive_utils.repos

WinstonHChou committed:
[Refactor] Use forked repo of Autodrive_dev_kit

WinstonHChou committed:
[BugFix] Joint state publisher

WinstonHChou committed:
[Feat] modified autodrive_racecar.xacro to given spec

WinstonHChou committed:
[Remap] remap imu and encoder topics

WinstonHChou committed:
[Refactor] No need for remapping encoders topics

WinstonHChou committed:
add update autodrive_devkit in makefile

WinstonHChou committed:
[Refactor] front wheels joint type changes to `continuous`
